optimization/@matRad_OptimizationProblemMixMod/matRad_constraintJacobian.m

```diff
@@ -194,31 +222,61 @@
-      mSqrtBetaDoseProjection{scenario} = sparse(voxelID,constraintID,mSqrtBetaDoseProjection{scenario},...
-         size(mAlphaDoseProjection{scenario},1),size(mAlphaDoseProjection{scenario},2));
-      
-      jacob   = mAlphaDoseProjection{scenario}' * dij.mAlphaDose{scenario} +...
-         mSqrtBetaDoseProjection{scenario}' * dij.mSqrtBetaDose{scenario};
-      
-   end
+bxidx = 1; %modality  bixel index
+
+for mod = 1: length(dij.original_Dijs)
+
+    
+    % enter if statement also for protons using a constant RBE
+    if isa(optiProb.BP,'matRad_DoseProjection')
+       
+       if ~isempty(DoseProjection{scenario})
+          jacob = [jacob,DoseProjection{scenario}' * dij.STfractions{mod} * dij.original_Dijs{mod}.physicalDose{scenario}];
+       end
+       
+    elseif isa(optiProb.BP,'matRad_ConstantRBEProjection')
+       
+       if ~isempty(DoseProjection{scenario})
+          jacob = [jacob,DoseProjection{scenario}' * dij.STfractions{mod}*  dij.original_Dijs{mod}.RBE * dij.original_Dijs{mod}.physicalDose{scenario}];
+       end
+       
+    elseif isa(optiProb.BP,'matRad_EffectProjection')
+       
+       if ~isempty(mSqrtBetaDoseProjection{scenario}) && ~isempty(mAlphaDoseProjection{scenario})
+
+          STrepmat = (~dij.spatioTemp(mod) + dij.spatioTemp(mod)*dij.numOfSTscen(mod));
+          wt = reshape(w(bxidx: bxidx+STrepmat*dij.original_Dijs{mod}.totalNumOfBixels-1),[dij.original_Dijs{mod}.totalNumOfBixels,STrepmat]);
+            
+          mSqrtBetaDoseProjectionmod = mSqrtBetaDoseProjection{scenario}' * dij.original_Dijs{mod}.mSqrtBetaDose{scenario} * wt;
+          mSqrtBetaDoseProjectionmod = sparse(voxelID,constraintID,mSqrtBetaDoseProjectionmod,...
+             size(mAlphaDoseProjection{scenario},1),size(mAlphaDoseProjection{scenario},2));
+          
+          jacob   = [jacob,mAlphaDoseProjection{scenario}' * dij.STfractions{mod}* dij.original_Dijs{mod}.mAlphaDose{scenario} +...
+             mSqrtBetaDoseProjectionmod' * dij.STfractions{mod}* dij.original_Dijs{mod}.mSqrtBetaDose{scenario}];
+
+          bxidx = bxidx + STrepmat*dij.original_Dijs{mod}.totalNumOfBixels;
+       end
+    end
+end
+jacobianChecker = 1;
+if jacobianChecker == 1
+            c =  matRad_constraintFunctions(optiProb,w,dij,cst);
+    epsilon = 1e-6;
+    ix2 = unique(randi([1 numel(w)],1,5));
+    ix = find((full(jacob)>1e-3),5, "first")';
+    for i=ix
+        wInit = w;
+        wInit(i) = wInit(i) + epsilon;
+        cDel= matRad_constraintFunctions(optiProb,wInit,dij,cst);
+        numJacob = (cDel - c)/epsilon;
+        fullJacob = full(jacob);
+        diff = (numJacob./fullJacob(:,i) - 1)*100;
+        fprintf(['grad val #' num2str(i) ' - rel diff numerical and analytical jacob = ' num2str(diff(1)) '\n']);
+        %fprintf([' any nan or zero for photons' num2str(sum(isnan(glog{1}))) ',' num2str(sum(~logical(glog{1}))) ' for protons: ' num2str(sum(isnan(glog{2}))) ',' num2str(sum(~logical(glog{2}))) '\n']);
```

Comment:
can you clean up these comments in the files and write description for the Jacobian checker and set its flag to 0